rllib/utils/metrics/stats/ema.py

```diff
@@ -0,0 +1,256 @@
+import logging
+from typing import Any, Dict, List, Union
+
+import numpy as np
+
+from ray.rllib.utils.framework import try_import_tf, try_import_torch
+from ray.rllib.utils.metrics.stats.base import StatsBase
+from ray.rllib.utils.metrics.stats.utils import safe_isnan, single_value_to_cpu
+from ray.util import log_once
+from ray.util.annotations import DeveloperAPI
+
+logger = logging.getLogger(__name__)
+
+torch, _ = try_import_torch()
+_, tf, _ = try_import_tf()
+
+
+@DeveloperAPI
+class EmaStats(StatsBase):
+    """A Stats object that tracks the exponential average of a series of values."""
+
+    stats_cls_identifier = "ema"
+
+    def __init__(
+        self,
+        ema_coeff: float = 0.01,
+        *args,
+        **kwargs,
+    ):
+        """Initializes a EmaStats instance.
+
+        We calculate the EMA in parallel components.
+        Also, we potentially aggregate them multiple times per reduction cycle.
+        We therefore aggregate by taking the mean of all collected EMAs.
```

Comment:
Could it make sense to set Stats at leafs to `item_sequence` and at the root have an `EmaStats` that ema-averages over all of them?